src/squelch.cpp

```diff
@@ -84,12 +84,35 @@ void Squelch::set_squelch_snr_threshold(const float &db) {
 				using_manual_level_ ? "true" : "false", normal_signal_ratio_, flappy_signal_ratio_, moving_avg_cap_);
 }
 
+void Squelch::set_ctcss_freq(const float &ctcss_freq, const float &sample_rate) {
+	// create two CTCSS detectors with different window sizes.  0.4 sec is required to tell between all the "standard"
+	// tones but 0.05 is enough to tell between tones ~20 Hz appart.  Will use ctcss_fast_ until there are enough samples
+	// for ctcss_slow_
+	ctcss_fast_ = CTCSS(ctcss_freq, sample_rate, sample_rate * 0.05);
+	ctcss_slow_ = CTCSS(ctcss_freq, sample_rate, sample_rate * 0.4);
+}
+
 bool Squelch::is_open(void) const {
-	return (current_state_ == OPEN || current_state_ == CLOSING);
+	if (current_state_ != OPEN && current_state_ != CLOSING) {
+		return false;
+	}
+	
+	if (ctcss_fast_.is_enabled()) {
+		if (ctcss_slow_.enough_samples()){
+			return ctcss_slow_.has_tone();
+		}
+		return ctcss_fast_.has_tone();
+	}
+	
+	return true;
```

Comment:
This change introduces a bug in scanning mode.  Without this change a state of `OPENING` results in `is_open()` returning `false`, with this change it returns `true`.  The impact is a scanning SDR will be stuck on